deploy/app.py

```diff
@@ -1,17 +1,38 @@
 #!/usr/bin/env python3
 import os
+import subprocess
+from getpass import getuser
 
-from aws_cdk import core
+import aws_cdk
+from aws_cdk import App, Environment
 
 from cdk.lambda_stack import LambdaStack
 from cdk.step_function_stack import StepFunctionStack
 from cdk.queue_stack import QueueStack
+from cdk.util_stack import DataPipelineUtilStack
 
-import config
+import config as cfg
 
-app = core.App()
+app = App()
 
-env_details = core.Environment(
+config = cfg.Config(_env_file=".env")
+
+git_sha = subprocess.check_output(["git", "rev-parse", "HEAD"]).decode().strip()
+try:
+    git_tag = subprocess.check_output(["git", "describe", "--tags"]).decode().strip()
+except subprocess.CalledProcessError:
+    git_tag = "no-tag"
```

Comment:
Very nice. Definitely superior to the sha-only strategy I've used in the past for this kind of thing